zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -316,50 +342,138 @@ benchmarks! {
+        assert!(current_len - 1 == prev_len);
     }
 
-    do_reject_market {
-        let (_, market_id) = create_market_common::<T>(
-            MarketCreation::Advised,
-            MarketType::Categorical(T::MaxCategories::get()),
+    deploy_swap_pool_for_market_open_pool {
+        let a in (T::MinCategories::get().into())..T::MaxCategories::get().into();
+
+        let (caller, market_id) = create_market_common::<T>(
+            MarketCreation::Permissionless,
+            MarketType::Categorical(a.saturated_into()),
             ScoringRule::CPMM
         )?;
+
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            // We need to ensure, that period range start is now, because we would like to open the pool now
+            market.period = MarketPeriod::Timestamp(T::MarketCommons::now()..T::MaxSubsidyPeriod::get());
+            Ok(())
+        })?;
+
         let market = T::MarketCommons::market(&market_id.saturated_into()).unwrap();
-    }: { Pallet::<T>::do_reject_market(&market_id, market)? }
+
+        let max_swap_fee: BalanceOf::<T> = MaxSwapFee::get().saturated_into();
+        let min_liquidity: BalanceOf::<T> = MinLiquidity::get().saturated_into();
+        let _ = Call::<T>::buy_complete_set { market_id, amount: min_liquidity }
+            .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
+
+        let weight_len: usize = MaxRuntimeUsize::from(a).into();
+        let weights = vec![MinWeight::get(); weight_len];
+
+        let call = Call::<T>::deploy_swap_pool_for_market { market_id, swap_fee: max_swap_fee, amount: min_liquidity, weights };
+    }: {
+        call.dispatch_bypass_filter(RawOrigin::Signed(caller).into())?;
+    } verify {
+        let market_pool_id = T::MarketCommons::market_pool(&market_id.saturated_into()).unwrap();
+        let pool = T::Swaps::pool(market_pool_id).unwrap();
+        assert!(pool.pool_status == PoolStatus::Active);
+    }
+
+    dispute_authorized {
+        let d in 0..(T::MaxDisputes::get() - 1) as u32;
+
+        let report_outcome = OutcomeReport::Scalar(u128::MAX);
+        let (caller, market_id) = create_close_and_report_market::<T>(
+            MarketCreation::Permissionless,
+            MarketType::Scalar(0u128..=u128::MAX),
+            report_outcome,
+        )?;
+
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            let admin = account("admin", 0, 0);
+            market.dispute_mechanism = MarketDisputeMechanism::Authorized(admin);
+            Ok(())
+        })?;
+
+        for i in 0..d {
+            let outcome = OutcomeReport::Scalar(i.into());
+            let disputor = account("disputor", i, 0);
+            let _ = T::AssetManager::deposit(Asset::Ztg, &disputor, (u128::MAX).saturated_into());
+            let _ = Call::<T>::dispute { market_id, outcome }.dispatch_bypass_filter(RawOrigin::Signed(disputor).into())?;
```

Comment:
There is no strong reason why I used the long syntax. I would say it was a copy and paste.

Reply:
87ceb6f56d82ccd44b9d44c693bfce85d0c7f9bd

Reply:
I see lots of things are unwarp() now, in certain cases we may not need Result<> return type. Although not required to change.

Reply:
I see your point. I think Malte suggested it, because it's much clearer to see first, see the error with just tracking `unwrap()` and second, to get a clearer message in case a call fails.